app/views/spree/admin/overview/single_enterprise_dashboard.html.haml

```diff
@@ -40,11 +40,18 @@
         %span.icon-user
         = t "your_profil_live"
     .list
-      %a.button.bottom{href: main_app.enterprise_shop_url(@enterprise), target: '_blank'}
-        = t "see"
-        = @enterprise.name
-        = t "live"
-        %span.icon-arrow-right
+      - if @enterprise.sells == "none"
+        %a.button.bottom{href: main_app.producers_url, target: '_blank'}
+          = t "see"
+          = @enterprise.name
+          = t "live"
+          %span.icon-arrow-right
+      - else
+        %a.button.bottom{href: main_app.enterprise_shop_url(@enterprise), target: '_blank'}
+          = t "see"
+          = @enterprise.name
+          = t "live"
+          %span.icon-arrow-right
```

Comment:
Nice! There's some duplicate code though. You could define a helper like this:
```rb
def enterprise_url(enterprise)
  if enterprise.is_distributor
    main_app.enterprise_shop_url(enterprise)
  else
    main_app.producers_url
  end
end
```

You can then use this helper in the view:
```suggestion
      %a.button.bottom{href: enterprise_url(@enterprise), target: '_blank'}
        = t "see"
        = @enterprise.name
        = t "live"
        %span.icon-arrow-right
```